site/content/entry/citra-progress-report-2017-august.md

```diff
@@ -0,0 +1,159 @@
+in software:
+
+{{% table %}}
+|      Feature      |                        Hardware                       |                                                   Software                                                   |
+|:-----------------:|:-----------------------------------------------------:|:------------------------------------------------------------------------------------------------------------:|
+| Fragment Lighting | [#1264](https://github.com/citra-emu/citra/pull/1264) |  [#2766](https://github.com/citra-emu/citra/pull/2766) [#2822](https://github.com/citra-emu/citra/pull/2822) |
+|   Spot Lighting   | [#2727](https://github.com/citra-emu/citra/pull/2727) |                             [#2871](https://github.com/citra-emu/citra/pull/2871)                            |
+|  Geometric Factor | [#2776](https://github.com/citra-emu/citra/pull/2776) |                             [#2872](https://github.com/citra-emu/citra/pull/2872)                            |
+|    Bump Mapping   | [#2762](https://github.com/citra-emu/citra/pull/2762) |                             [#2891](https://github.com/citra-emu/citra/pull/2891)                            |
+{{% /table %}}
+
+Although most users won't benefit much from this in the present, it is important
+to have a complete software implementation of the 3DS' GPU so that Citra can be
+used as a stable, working, and complete reference implementation in the future,
+when obtaining a working real console may be much more difficult.
+
+## [Handle Invalid Filenames When Renaming](https://github.com/citra-emu/citra/pull/2850) by [j-selby](https://github.com/j-selby)
+
+Citra emulates the 3DS system services at a high level of emulation, or HLE for
+short. What this means is that every time a 3DS application or game running in
+Citra makes a request to the 3DS System Software, Citra captures the request and
+tries to translate it into its PC equivalent, rather then running the 3DS System
+Software directly.
+
+For example, if a game makes a call to [`FS:OpenFile`](https://www.3dbrew.org/wiki/FS:OpenFile),
+Citra in turn calls the operating system's file opening function
+([`_wfopen_s()`](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/fopen-s-wfopen-s) on Windows,
+or [`open()`](http://pubs.opengroup.org/onlinepubs/9699919799/functions/open.html) on macOS and Linux),
+with [the path to the virtual SD card](/wiki/user-directory/) added to the beginning.
+
+Now, on top of the usual names for files and folders, there's two special folders
+inside every single folder on your computer, `.` and `..`. These aren't actual
+folders in the sense that you can place files and other folders inside of them.
+Instead, they each symbolize the current folder, and the folder one level above
+it, respectively. For example, `C:/Windows/System32/..` actually means `C:/Windows/`.
+
+The reason I mention this, is because a few Citra developers believed a game could
```

Comment:
"I" got thrown in here outta nowhere. I suggest a different lead in phrase like "With this background information in mind, a few Citra developers..."